tooling/bundler/src/bundle/windows/templates/installer.nsi

```diff
@@ -135,7 +135,13 @@ Function PageReinstall
     ReadRegStr $R0 HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$1" "DisplayName"
     ReadRegStr $R1 HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$1" "Publisher"
     StrCmp "$R0$R1" "${PRODUCTNAME}${MANUFACTURER}" 0 wix_loop
-    StrCpy $R5 "wix"
+    ClearErrors
+    ReadRegStr $2 HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$1" "UninstallString"
+    Push $2
+    Call GetFirstStrPart
+    Pop "$R0"
+    StrCmp $R0 "MsiExec.exe" 0 wix_done
+    StrCpy $R7 "wix"
```

Comment:
Instead of a custom function, there is built-in functions https://nsis.sourceforge.io/StrFunc to manipulate strings

So add at the top

```nsi
!include "StrFunc.nsh"
${StrCase}
${StrLoc}
```

and then
```nsi
ReadRegStr $R0 HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$1" "UninstallString"
${StrCase} $R1 $R0 "L"
${StrLoc} $R0 $R1 "msiexec" ">"
StrCmp $R0 0 0 wix_done
StrCpy $5 "wix"
```

Reply:
```
Unicode true
SetCompressor /SOLID lzma
```
I need to move these two lines to the head of the file.

Reply:
why so?